js/pages/tola_management_pages/user/components/edit_user_programs.js

```diff
@@ -33,10 +32,10 @@ const create_program_objects = (programs, store) => Object.entries(programs)
                                                            }),{})
 
 //we need to flatten the country -> program heirarchy to support the virtualized table
-const flattened_listing = (countries, programs) => countries.flatMap(country =>
+const flattened_listing = (countries, programs, isExpanded) => countries.flatMap(country =>
```

Comment:
`isExpanded` is a bit opaque to me. Where does this value come from? Is it toggled when I click the expando?

Reply:
I pass it in whenever it's called, because the specs broke this functional design and required a stateful element to determine if its expanded (a combination of various filter states), so I bound a function to the required states and passed it in each time flattened_listing is called.  Would a @param jsdoc type function docstring help?  Or should I just refactor that whole flattened_listing to be a method on the store (so it has access to internal state)?

Reply:
a docstring would probably do it, I mean I can tell what it should be telling me (`isExpanded`, duh). I think this + the `...` syntax is kind of confusing to me, like: `flattened_programs` is a 1-dimensional array (of indeterminate length), of n countries, n programs, and the value of `isExpanded`?

Forgive my ignorance here, js just makes my eyes water

Reply:
>  js just makes my eyes water

you and me both.  Tried adding some jsdoc explaining the paradigm on that flattened listing function and refactored the calls to it so the callback function is more clearly shown (and commented)